source/memory/memory.go

```diff
@@ -113,6 +129,29 @@ func (s *memorySource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
+// detectSwap detects Swap node information
+func detectSwap() (map[string]string, error) {
+	procBasePath := hostpath.ProcDir.Path("swaps")
+	file, err := os.Open(procBasePath)
+	defer file.Close()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think CI is yelling because of this:
```suggestion
	if err != nil {
		return nil, err
	}
	defer file.Close()
```

Reply:
addressed.